packages/ember-routing/lib/system/generate_controller.js

```diff
@@ -17,7 +17,7 @@ import {
 */
 
 export function generateControllerFactory(owner, controllerName, context) {
-  let Factory = owner._lookupFactory('controller:basic').extend({
+  let Factory = owner.factoryFor('controller:basic').class.extend({
```

Comment:
@chadhietala and I went over this. The change as it stands would indeed squash DI in the case where there was an injection on `controller:basic` and you expected it to show up on `controller:generated-controller`. Of course that whole concept, that your DI onto `controller:basic` will show up on `controller:generated-controller`, is pretty WAT. But we should preserve it if possible.